### DISCUSSION TITLE
Configure Sublime 3 workspace settings

### DISCUSSION THREAD

**Discussion by luciansava** [wprig/wprig#689] [Q&A] [answered]
Hello!
How should one Configure workspace settings https://wprig.io/learn/configure-vs-code-workspace-settings-v2/
for Sublime 3?
(I already installed necessary extensions) 
Thank you!

p.s.: I hope this is proper place for asking thins question, if not, please asccept my appoligies

**Comment by jacklowrie** [comment, marked as answer]
You're in the right place! 

I just switched back to Sublime from Atom recently, and am using this question as an excuse to (finally) finish setting up my linters. So we might have to figure some of this out together. There also may be some sublime packages equivalent to the VS Code-specific ones in the docs that I don't know about.

I've got [SublimeLinter](http://www.sublimelinter.com/en/stable/), with the [eslint](https://github.com/SublimeLinter/SublimeLinter-eslint), [stylelint](https://github.com/SublimeLinter/SublimeLinter-stylelint), and [phpcs](https://github.com/SublimeLinter/SublimeLinter-phpcs) plugins.

Here's the linter settings from my `.sublime-project` for working on wprig (set up today):

```json
{
	"folders":
	[
		{
			"path": "."
		}
	],

    "settings": {
        "SublimeLinter.linters.phpcs.lint-mode": "background",
        "SublimeLinter.linters.phpcs.executable": "${folder}/vendor/bin/phpcs",
        "SublimeLinter.linters.eslint.lint-mode": "background",
        "SublimeLinter.linters.eslint.env": {"PATH":"/Users/jack/.nvm/versions/node/v13.8.0/bin/"},
	"SublimeLinter.linters.stylelint.lint-mode": "background",
	"SublimeLinter.linters.stylelint.executable": "${folder}/node_modules/.bin/stylelint",
	"SublimeLinter.linters.stylelint.env": {"PATH":"/Users/jack/.nvm/versions/node/v13.8.0/bin/"},
    }
}
```

A couple notes:
- `lint-mode` can be set in your SublimeLinter settings rather than in the project workspace, unless you prefer to leave  on `manual` or another setting by default.
- I noticed that for me, ESLint and stylelint weren't working because for some reason the linters couldn't find Node - adding the path as a project-specific setting for each linter fixed that for me, but your path to Node may be different. found that in the [SublimeLinter troubleshooting docs](http://www.sublimelinter.com/en/latest/troubleshooting.html#debugging-path-problems). This could also be set in SublimeLinter settings rather than in each project.

Let me know if this works for you, and if you find any other tricks/tweaks for using SublimeText & WP Rig!

**Reply by luciansava** [reply]
Thank you for your reply, Jack, I'll give it another try from scratch and let you know if works. at a first look, I cand see that i don't have a .nvm folder in /Users/MyWindowsUserName/.nvm/versions/node/v13.8.0/bin/

**Reply by luciansava** [reply]
started again, from the begining. 
followed the tuts from here https://wprig.io/learn/run-the-wp-rig-installer-v2/
managed to Run the WP Rig installer succesfully. 

with VScode editor, this time.

**Reply by luciansava** [reply]
after npm run dev, 2 problems:
1. all php files are listed with errors PHP Code Sniffer found a problem in C:\Users\Solarian\Local Sites\wprig\app\public\wp-content\themes\wprig\404.php
2. takes pretty long time to relaod the browser. Even to open Hello world post it takes like forever

**Reply by luciansava** [reply]
The following tasks did not complete: generateCert
Failed at the wprig@2.0.1 generateCert script.

**Reply by dthenley** [reply]
> 1. all php files are listed with errors PHP Code Sniffer found a problem in C:\Users\Solarian\Local Sites\wprig\app\public\wp-content\themes\wprig\404.php

I ran into an issue similar to this. Does this happen when you download the zip file, unzip it and then compile it? 

Also what are the PHP code sniffer errors are you getting?

**Reply by luciansava** [reply]
Hi, @dthenley!
I used the zip from here https://github.com/wprig/wprig/tree/v2.0
as seen here https://wprig.io/learn/get-a-fresh-copy-of-wp-rig-v2/ second 12, bottom right screen in video
had the same result with latest https://github.com/wprig/wprig/tree/2.0.2

Also what are the PHP code sniffer errors are you getting? 
I am sure about this. I can't replicate right now as I deleted all, waiting for some spare time for retesting. 
WPRIG is a tool that promise a lot, I am sure that at some point it will deliver :)
